bedrock/settings/__init__.py

```diff
@@ -127,6 +127,7 @@
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["object-src"] = [csp.constants.NONE]
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["frame-ancestors"] = [csp.constants.NONE]
     CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["style-src"].remove(csp.constants.UNSAFE_INLINE)
+    CONTENT_SECURITY_POLICY_REPORT_ONLY["DIRECTIVES"]["form-action"] = [csp.constants.SELF, BASKET_URL, FXA_ENDPOINT]
```

Comment:
**question:** FXA_ENDPOINT has a trailing slash and BASKET_URL does not. Is that gonna cause any wrinkles or is CSP smart enough to strip trailing slashes?

Reply:
Thanks for noticing that. The browser, from what I've read, doesn't really differentiate, so this should be fine. But I'm a fan of consistency so perhaps I'll add some trailing slash stripping.

Reply:
NB: The slash has a function in CSP wrt to being a path component (i.e. an implicit wildcard after it) or not.

Reply:
ref: https://w3c.github.io/webappsec-csp/#path-part-match
see: https://bugzilla.mozilla.org/show_bug.cgi?id=741019#c25

Reply:
Looking at the CSP `path-part-match` rules more closely...

1. With `https://example.com/` in our CSP, a form submit to `https://example.com/` matches.
2. With `https://example.com` in our CSP (no path), a form submit to `https://example.com/` matches, since when split on `'/'`, the path is empty string.

And vice-versa for the form submit to the non-slash URL.

Focusing on the `BASKET_URL`, however, reading the `path-part-match` rules @janbrasna mentioned, we would want the trailing slash for the basket URL. Without the trailing slash it is in "exact match" mode and a form submit to an actual path would fail. In the comment above I was just looking at the `FXA_ENDPOINT` in our code where we don't use any path.

So the slash is important, just not necessarily if we're submitting to `""` vs `/`. 

@janbrasna - does that "match" your understanding of it?

Reply:
I'm working on adding some more basket APIs under `/api/`. So I'd probably keep that one at the root level for now. Then again, the ones under `/api/` are more for javascript, not form submits.

Reply:
Looking at this in context of #15787 — even if that's a JSON-only API, it still formally uses form's action; so thinking about how important It would be to support submitting that form (in nonJS, or in case the JS loading fails etc.; to still allow users to recover(?)…)